### DISCUSSION TITLE
Keep order of terms

### DISCUSSION THREAD

**Discussion by DaniMarts** [sympy/sympy#21065] [Q&A]
Is it possible to change a something in sympy to make the order of terms be the same as the one inputted by the user? In some fields, there is a convention on the order of terms, and not following this convention can make the document harder to read.

If so, has anyone been able to do it?

**Comment by oscarbenjamin** [comment]
There are two separate steps to this. The first is to make sure that the internal representation does not reorder the terms which you can do using `evaluate=False`. The second part is to make sure that the printing code does not reorder the terms and I'm not sure if that's currently possible.

**Reply by DaniMarts** [reply]
In which function do I use `evaluate=False`? 
I tried using `evaluate=False` in the following code snippet, 
``` 
x, y, z = symbols("x y z")
srepr(Eq(y, sin(z - x), evaluate=False))
```
,
but with no success, as I got "y = - sin(x - z)" (_"Equality(Symbol('y'), Mul(Integer(-1), sin(Add(Symbol('x'), Mul(Integer(-1), Symbol('z'))))))"_ ) as a result, instead of "y = sin(z - x)", which is what I expected.

**Reply by oscarbenjamin** [reply]
It should be `sin(Add(z, -x, evaluate=False), evaluate=False)`

**Reply by DaniMarts** [reply]
That is better, as the result will be sin(-x+z), instead of -sin(x-z). But is it possible to make the result sin(z-x), rather? I know the [sympy website](https://docs.sympy.org/latest/tutorial/manipulation.html) says :

> This because in SymPy, the arguments of the commutative operations Add and Mul are stored in an arbitrary (but consistent!) order, which is independent of the order inputted (if you’re worried about noncommutative multiplication, don’t be. In SymPy, you can create noncommutative Symbols using Symbol('A', commutative=False), and the order of multiplication for noncommutative Symbols is kept the same as the input). 

So I was really asking if there was a way to create "_noncommutative addition_" in sympy, as it seems to always prefer to sort the symbols in alphabetical order, not caring if they must follow a certain convention. I know mathematically, the expression is the same, regardless of the order.

**Reply by oscarbenjamin** [reply]
The problem is in the printing code that displays the result. You can check the order of the args in internal representation:
```
In [12]: sin(Add(z, -x, evaluate=False), evaluate=False).args[0].args
Out[12]: (z, -x)

In [13]: sin(Add(z, -x, evaluate=False), evaluate=False)
Out[13]: sin(-x + z)
```

**Reply by goekce** [reply]
If the printing code would not reorder the symbols, this would ease the comparison of an equation with another source.

The documentation talks about a method for avoiding reordering for multiplication here <https://docs.sympy.org/latest/tutorials/intro-tutorial/manipulation.html#understanding-expression-trees>

```py
a, b = symbols('a b')
b * a  # Result `ba`

a, b = symbols('a b', commutative=False)
b * a  # Result `ba`
b + a  # Result a + b
```
Unfortunately, `commutative=False` applies only to multiplication.

I looked at the pretty printing API but found no valuable parameter to keep the order.
